protocol/blockchain/cryptography/multisignature.md

```diff
@@ -0,0 +1,84 @@
+
+```
+<dummy> <sig1> ... <sigM>
+```
+
+Upon script execution, this will act like:
+
+```
+<dummy> <sig1> ... <sigM> M <pubkey1> ... <pubkeyN> N CHECKMULTISIG
+```
+
+Due to a historical bug (the original implementation of `OP_CHECKMULTISIG` consumed `M+N+3` items on the stack instead of `M+N+2`), an extra unusued value called the `dummy` element was included in the script. This was usually done via `OP_0`. Since the [HF-2019115](/protocol/forks/hf-20191115) Bitcoin Cash upgrade, this element has been repurposed as a trigerring and execution mechanism. 
+
+In particular, the value of the `dummy` element determines whether ECDSA or Schnorr signatures have to be used. If `dummy` is `0` (i.e., an empty byte array), then all signatures must be produced by ECDSA. If `dummy` is not `0`, then all signatures must be produced by the Schnorr algorithm and the `dummy` element is interpreted as a bitfield called `checkbits`.
+
+Upon `OP_CHECKMULTISIG` execution, the ECDSA mode (`dummy = 0`) is:
+
+* The first signature is checked against each public key until a match is found.
+* Starting with the subsequent public key, the second signature is checked against each remaining public key until a match is found.
+* The process is repeated until all signatures have been checked or not enough public keys remain to produce a successful result.
+* All signatures need to match a public key, otherwise `false` is returned.
+
+The Schnorr mode (`dummy = checkbits > 0`) operates similarly, but only checks signatures as requested, according to the `checkbits` field:
+
+* If the least significant bit of `checkbits` is set, then the first signature is checked against the first public key. 
+* If it is not but if the second least significant bit is set, then the first signature is checked against the second public key. 
+* Once the first signature has been checked against a matching public key, `checkbits` is bit-shifted to the right (`checkbits := checkbits >> 1`).
+* The process is repeated until all signatures have been checked or not enough public keys remain to produce a successful result.
+* All signatures need to match a public key, otherwise `false` is returned.
+* If the final `checkbits` value is non-zero, then `false` is returned. 
+
+Because public keys are not checked again if they fail any signature comparison (in both cases), signatures must be placed in the unlocking script using the same order as their corresponding public keys were placed in the locking script (P2MS) or redeem script (P2SH).
+
+Note that the `checkbits` element is encoded as a byte array of length `floor((N + 7)/8)` (the shortest byte array that can hold N bits) and must have exactly M bits set to ensure that all signatures are checked against public keys.
+
+To know more about the Schnorr mode, see the [official specification](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2019-11-15-schnorrmultisig.md).
```

Comment:
I don't think the term "official" anything exists for the Bitcoin Cash protocol, so just "specification" is better IMO. (Also we want to add a copy of it in this repo)

Reply:
Yes I agree.